delta/rdf/src/test/scala/ch/epfl/bluebrain/nexus/delta/rdf/jsonld/decoder/JsonLdDecoderSpec.scala

```diff
@@ -0,0 +1,98 @@
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.JsonLd
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.{JsonLdContext, RemoteContextResolution}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder.JsonLdDecoderError.DecodingDerivationFailure
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder.JsonLdDecoderSpec.Drink.{Cocktail, Volume}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder.JsonLdDecoderSpec.Menu.DrinkMenu
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder.JsonLdDecoderSpec.{Drink, Menu}
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder.configuration.semiauto._
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder.semiauto._
+import ch.epfl.bluebrain.nexus.delta.rdf.syntax._
+import ch.epfl.bluebrain.nexus.testkit.{CirceLiteral, EitherValuable, IOValues, TestHelpers}
+import org.scalatest.Inspectors
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpecLike
+
+import scala.annotation.nowarn
+
+@nowarn("cat=unused")
+class JsonLdDecoderSpec
+    extends AnyWordSpecLike
+    with Matchers
+    with Inspectors
+    with CirceLiteral
+    with EitherValuable
+    with IOValues
+    with TestHelpers {
+
+  implicit private val rcr: RemoteContextResolution = RemoteContextResolution.fixed()
+
+  "A JsonLdDecoder" should {
+
+    val json                                          = jsonContentOf("/jsonld/decoder/cocktail.json")
+    val jsonLd                                        = JsonLd.expand(json).accepted
+    val context                                       = jsonContentOf("/jsonld/decoder/context.json")
+    val ctx                                           = JsonLdContext(context.topContextValueOrEmpty).accepted
+    implicit val config: Configuration                = Configuration.default.copy(context = ctx)
+    implicit val volumeDecoder: JsonLdDecoder[Volume] = deriveConfigJsonLdDecoder[Volume]
+
+    "decode a Menu" in {
+      implicit val drinkDecoder: JsonLdDecoder[Drink] = deriveConfigJsonLdDecoder[Drink]
+      implicit val menuDecoder: JsonLdDecoder[Menu]   = deriveConfigJsonLdDecoder[Menu]
+
+      jsonLd.to[Menu].rightValue shouldEqual
+        DrinkMenu(
+          Set(
+            Cocktail(
+              schema + "mojito",
+              alcohol = true,
+              ing = Set("rum", "sugar"),
+              name = "Mojito",
+              steps = NonEmptyList.of("cut", "mix"),
+              volume = Volume("%", 8.3),
+              link = Some(schema + "mylink"),
+              instant = Instant.parse("2020-11-25T21:29:38.939939Z")
+            )
+          )
+        )
+    }
+
+    "fail decoding a Menu" in {
+      implicit val drinkDecoder: JsonLdDecoder[Drink] = deriveJsonLdDecoder[Drink]
+      implicit val menuDecoder: JsonLdDecoder[Menu]   = deriveJsonLdDecoder[Menu]
+      jsonLd.to[Menu].leftValue shouldBe a[DecodingDerivationFailure]
+    }
+  }
+
+}
+object JsonLdDecoderSpec {
```

Comment:
I think we're missing some tests for some instances...

Reply:
Yes, FiniteDuration, UUID and so on